src/app/components/core/configuration-manager/list-type-configuration/list-type-configuration.component.ts

```diff
@@ -69,8 +102,48 @@ export class ListTypeConfigurationComponent implements OnInit {
           return num;
         });
       }
-      this.form.get(this.configuration.key)?.patchValue(JSON.stringify(value))
-      this.formState.emit(this.listItems.valid);
+      if(this.configuration.items == 'object') {
+        let listValues = this.generateListValuesArray(value);
+        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(listValues) });
+      }
+      else{
+        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(value) });
+      }
```

Comment:
```
if(this.configuration.items == 'object') {
        let listValues = this.generateListValuesArray(value);
        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(listValues) });
      }
      else{
        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(value) });
      }
```

Above code block can be optimized to

```
if(this.configuration.items == 'object') {
    value = this.generateListValuesArray(value);
}
this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(value) });
```

Also why JSON.stringify in 
`this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(value) });` ?

Reply:
We need to send this in string otherwise in payload it is sent as array and API throws error.